src/go/rucio-dataset-mon-go/detailed_datasets/detailed_datasets.go

```diff
@@ -0,0 +1,112 @@
+package detailed_datasets
+
+import (
+	"context"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/models"
+	mymongo "github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/mongo"
+	"github.com/dmwm/CMSMonitoring/src/go/rucio-dataset-mon-go/utils"
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"log"
+)
+
+var (
+	collectionName = "detailed_datasets"
+	collection     *mongo.Collection
+	ProdAccounts   = []string{"transfer_ops", "wma_prod", "wmcore_output", "wmcore_transferor", "crab_tape_recall", "sync"}
+)
+
+// CreateSortBson creates sort object which support multiple column sort
+func CreateSortBson(dtRequest models.DataTableCustomRequest) bson.D {
+	orders := dtRequest.Orders
+	columns := dtRequest.Columns
+	sortBson := bson.D{}
+	for _, order := range orders {
+		intSortDirection := utils.ConvertOrderEnumToMongoInt(order.Dir)
+		if intSortDirection != 0 {
+			sortBson = append(sortBson, bson.E{
+				Key:   columns[order.Column].Data, // column name
+				Value: intSortDirection,           // column direction as int value (0/1)
+			})
+		}
+	}
+	return sortBson
+}
+
+// CreateSearchBson creates search query
+func CreateSearchBson(dtRequest models.DataTableCustomRequest) bson.M {
+	findQuery := bson.M{}
+
+	customs := dtRequest.Custom
+	if customs.Dataset != "" {
+		findQuery["Dataset"] = primitive.Regex{Pattern: customs.Dataset, Options: "im"}
```

Comment:
I'm not sure about `primitite.Regex` but you should check if you compile regex each time you call the function or not. If the former, you should avoid that and create compiled regex expression only once to avoid wasting time on that on each function call.

Reply:
ok